tools/dotcms-cli/api-data-model/src/test/java/com/dotcms/common/SiteTestHelper.java

```diff
@@ -0,0 +1,99 @@
+package com.dotcms.common;
+
+import com.dotcms.api.SiteAPI;
+import com.dotcms.api.client.model.RestClientFactory;
+import com.dotcms.model.site.GetSiteByNameRequest;
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+
+public class SiteTestHelper {
+
+    @Inject
+    RestClientFactory clientFactory;
+
+    /**
+     * Checks if a site with the given name exists.
+     *
+     * @param siteName the name of the site to check
+     * @return true if the site exists, false otherwise
+     */
+    protected Boolean siteExist(final String siteName) {
+        final var siteAPI = clientFactory.getClient(SiteAPI.class);
+        return siteExist(siteName, siteAPI);
+    }
+
+    /**
+     * Checks if a site with the given name exists.
+     *
+     * @param siteName the name of the site to check
+     * @param siteAPI  the site api to use
+     * @return true if the site exists, false otherwise
+     */
+    protected Boolean siteExist(final String siteName, final SiteAPI siteAPI) {
+
+        long start = System.currentTimeMillis();
+        long end = start + 15 * 1000; // 15 seconds * 1000 ms/sec
+        while (System.currentTimeMillis() < end) {
+            try {
+                var response = siteAPI.findByName(
+                        GetSiteByNameRequest.builder().siteName(siteName).build()
+                );
+                if ((response != null && response.entity() != null) &&
+                        ((response.entity().isLive() != null &&
+                                response.entity().isLive()) &&
+                                (response.entity().isWorking() != null &&
+                                        response.entity().isWorking()))) {
+                    return true;
+                }
+            } catch (NotFoundException e) {
+                // Do nothing
+            }
+
+            try {
+                Thread.sleep(2000); // Sleep for 2 second
+            } catch (InterruptedException ex) {
```

Comment:
We need to get away from the anti pattern of using Thread.sleep in tests.   Should use the Awaitility library instead.  We have it in master. 

https://www.baeldung.com/java-thread-sleep-vs-awaitility-await

If changing is going to delay getting master working then look at this when you come back around.

Reply:
Removed `Thread.sleep`, now we are using `org.testcontainers.shaded.org.awaitility.Awaitility.await`.